airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -211,10 +215,28 @@ def __init__(self, api: API, start_date: str = None, **kwargs):
 
     @property
     def name(self) -> str:
-        stream_name = self.__class__.__name__
-        if stream_name.endswith("Stream"):
-            stream_name = stream_name[: -len("Stream")]
-        return stream_name
+        if self._name:
+            return self._name
+        return super().name
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        json_schema = super().get_json_schema()
+        if self.properties:
+            json_schema["properties"]["properties"] = {"type": "object", "properties": self.properties}
```

Comment:
Some entities has dynamic set of properties, so we trying to resolve those at runtime